add missing entries and release section to changelog in preparation for release (#3550)
grafana/mimir: CHANGELOG.md

## Grafana Mimir - main / unreleased

## 2.5.0

### Grafana Mimir

* [CHANGE] Flag `-azure.msi-resource` is now ignored, and will be removed in Mimir 2.7. This setting is now made automatically by Azure. #2682
* [CHANGE] Experimental flag `-blocks-storage.tsdb.out-of-order-capacity-min` has been removed. #3261
* [CHANGE] Distributor: Wrap errors from pushing to ingesters with useful context, for example clarifying timeouts. #3307
* [CHANGE] The default value of `-server.http-write-timeout` has changed from 30s to 2m. #3346
* [CHANGE] Reduce period of health checks in connection pools for querier->store-gateway, ruler->ruler, and alertmanager->alertmanager clients to 10s. This reduces the time to fail a gRPC call when the remote stops responding. #3168
* [CHANGE] Hide TSDB block ranges period config from doc and mark it experimental. #3518

> * [CHANGE] Chore: use upstream rules.SendAlerts(). #3271

Reviewer: What has changed from user's perspective? Why does Mimir user/admin care?

Reviewer 2: Nothing has changed from user perspective. We can remove it.

Author: Fixed: 45b55e81d

* [ENHANCEMENT] Store-gateway: improved performance of series matching. #3391
* [ENHANCEMENT] Move the validation of incoming series before the distributor's forwarding functionality, so that we don't forward invalid series. #3386 #3458
* [ENHANCEMENT] S3 bucket configuration now validates that the endpoint does not have the bucket name prefix. #3414

> * [ENHANCEMENT] Store-gateway: index stats response. #3206

Reviewer: What does this mean?

Reviewer 2: We can rewrite it as:

```suggestion
* [ENHANCEMENT] Query-frontend: added "fetched index bytes" to query statistics, containing the total bytes read by store-gateways from TSDB block indexes. #3206
```

Author: > The term "containing" is ambiguous: is it "that contain" or "so that the statistics contain" or something else?

The latter, I updated it accordingly: 45b55e81d

* [ENHANCEMENT] Distributor: push wrapper should only receive unforwarded samples. #2980
* [BUGFIX] Flusher: Add `Overrides` as a dependency to prevent panics when starting with `-target=flusher`. #3151
* [BUGFIX] Updated `golang.org/x/text` dependency to fix CVE-2022-32149. #3285
* [BUGFIX] Query-frontend: properly close gRPC streams to the query-scheduler to stop memory and goroutines leak. #3302
* [BUGFIX] Ruler: persist evaluation delay configured in the rulegroup. #3392
* [BUGFIX] Ring status pages: show 100% ownership as "100%", not "1e+02%". #3435
* [BUGFIX] Fix panics in OTLP ingest path when parse errors exist. #3538

> * [BUGFIX] Canonicalize orgid header in forwarding. #3385

Reviewer: I would just add this PR number next to 3283. This isn't a bugfix between releases since we started to include this header only in Mimir 2.5.0.

Author: Done: 45b55e81d

* [ENHANCEMENT] Dashboards: Remove the "Instance Mapper" row from the "Alertmanager Resources Dashboard". This is a Grafana Cloud specific service and not relevant for external users. #3152
* [ENHANCEMENT] Dashboards: Add "remote read", "metadata", and "exemplar" queries to "Mimir / Overview" dashboard. #3245
* [ENHANCEMENT] Dashboards: Use non-red colors for non-error series in the "Mimir / Overview" dashboard. #3246
* [ENHANCEMENT] Dashboards: Add support to multi-zone deployments for the experimental read-write deployment mode. #3256
* [ENHANCEMENT] Dashboards: If enabled, add new row to the `Mimir / Writes` for distributor autoscaling metrics. #3378
* [ENHANCEMENT] Dashboards: Add read path insights row to the "Mimir / Tenants" dashboard. #3326
* [ENHANCEMENT] Alerts: Add runbook urls for alerts. #3452
* [ENHANCEMENT] Configuration: Make it possible to configure namespace label, job label, and job prefix. #3482
* [ENHANCEMENT] Dashboards: improved resources and networking dashboards to work with read-write deployment mode too. #3497 #3504 #3519 #3531

> * [ENHANCEMENT] Alerts: add alert for high error rate in Distributor's forwarding feature. #3200

Reviewer 2: ```suggestion
* [ENHANCEMENT] Alerts: add "MimirDistributorForwardingErrorRate" alert, firing on high error rate in distributor's forwarding feature. #3200
```

Reviewer 3: We can tweak this a bit:
`* [ENHANCEMENT] Alerts: add "MimirDistributorForwardingErrorRate" alert, firing on high error rate in distributor's forwarding feature. #3200`

Potential alternative, given technically accurate:
`* [ENHANCEMENT] Alerts: Added "MimirDistributorForwardingErrorRate" alert, which fires on high error rates in the distributor’s forwarding feature. #3200`

> * [ENHANCEMENT] Dashboards: fix: use product name in overview dashboard instead of hardcoded "Mimir". #3209

Reviewer 2: Remove this. It's a no-op for Mimir users. This just affect GEM.

Author: Fixed: 9403c0e75d32c45e6ae446e2f11e20ce168ebb45

* [ENHANCEMENT] The query-tee node port (`$._config.query_tee_node_port`) is now optional. #3272
* [ENHANCEMENT] Add support for autoscaling distributors. #3378
* [ENHANCEMENT] Make auto-scaling logic ensure integer KEDA thresholds. #3512

> * [ENHANCEMENT] Refactor read-write-deployment jsonnet. #3405

Reviewer 2: Just add this PR number to the CHANGELOG entry "[FEATURE] Added support for experimental read-write deployment mode"

Author: Fixed: 9403c0e75d32c45e6ae446e2f11e20ce168ebb45

> * [ENHANCEMENT] Improve phrasing in Overview dashboard. #3488

Reviewer 2: Mixin changelog entries should be under "Mixin" section.

Author: Fixed: 9403c0e75d32c45e6ae446e2f11e20ce168ebb45

* [ENHANCEMENT] Documented how to configure HA deduplication using Consul in a Mimir Helm deployment. #2972
* [ENHANCEMENT] Improve `MimirQuerierAutoscalerNotActive` runbook. #3186
* [ENHANCEMENT] Improve `MimirSchedulerQueriesStuck` runbook to reflect debug steps with querier auto-scaling enabled. #3223

> * [ENHANCEMENT] Mimir 2.4 release notes. #3104 #3319

Reviewer 2: I would remove this. It was part of 2.4 release.

Author: Fixed: 9403c0e75d32c45e6ae446e2f11e20ce168ebb45

* [ENHANCEMENT] Use imperative for docs titles. #3178 #3332 #3343
* [ENHANCEMENT] Docs: mention gRPC compression in "Production tips". #3201
* [ENHANCEMENT] Update ADOPTERS.md. #3224 #3225
* [ENHANCEMENT] Add a note for jsonnet deploying. #3213
* [ENHANCEMENT] out-of-order runbook update with use case. #3253
* [ENHANCEMENT] Fixed TSDB retention mentioned in the "Recover source blocks from ingesters" runbook. #3280
* [ENHANCEMENT] Run Grafana Mimir in production using the Helm chart. #3072

> * [ENHANCEMENT] Use common config in the tutorial again. #3282

Reviewer 2: ```suggestion
* [ENHANCEMENT] Use common config in the tutorial. #3282
```

Reviewer 3: config --> configuration

Author: Fixed: 9403c0e75d32c45e6ae446e2f11e20ce168ebb45

> * [ENHANCEMENT] Clarify changelog and remove duplicate flag in the documentation. #3370

Reviewer 2: I would remove this.

* [ENHANCEMENT] Updated detailed steps for migrating blocks from Thanos to Mimir. #3290
* [ENHANCEMENT] Add scheme to DNS service discovery docs. #3450
* [BUGFIX] Remove reference to file that no longer exists in contributing guide. #3404
* [BUGFIX] Fix some minor typos in the contributing guide and on the runbooks page. #3418
* [BUGFIX] Fix small typos in API reference. #3526
* [BUGFIX] Fixed TSDB retention mentioned in the "Recover source blocks from ingesters" runbook. #3278
* [BUGFIX] Fixed configuration example in the "Configuring the Grafana Mimir query-frontend to work with Prometheus" guide. #3374

### Tools

* [FEATURE] Add `copyblocks` tool, to copy Mimir blocks between two GCS buckets. #3264
* [ENHANCEMENT] copyblocks: copy no-compact global markers and optimize min time filter check. #3268
* [ENHANCEMENT] Mimir rules GitHub action: Added the ability to change default value of `label` when running `prepare` command. #3236

> * [ENHANCEMENT] Use delve compatible with Go 1.19.2. #3166

Reviewer: This isn't user-facing. It only updated docker-compose dev setup under `./development/`.

Author: Done: 45b55e81d
